java/api/Transaction.java

```diff
@@ -67,6 +67,20 @@ public interface Transaction extends AutoCloseable {
     @CheckReturnValue
     Promise<? extends QueryAnswer> query(String query) throws com.typedb.driver.common.exception.TypeDBDriverException;
```

Comment:
I believe you could make this a `default` method that calls the one with options (with default options) so you don't have to implement twice?